keadm/cmd/keadm/app/cmd/ctl/restart/pod.go

```diff
@@ -71,51 +69,57 @@ func AddRestartPodFlags(cmd *cobra.Command, RestartPodOptions *PodRestartOptions
 }
 
 func (o *PodRestartOptions) restartPod(podNames []string) error {
-	for _, podName := range podNames {
-		podRequest := &restful.PodRequest{
-			Namespace: o.Namespace,
-			PodName:   podName,
-		}
-		pod, err := podRequest.GetPod()
-		if err != nil {
-			fmt.Println(err.Error())
-			continue
-		}
-		config, err := util.ParseEdgecoreConfig(common.EdgecoreConfigPath)
-		if err != nil {
-			fmt.Printf("get edge config failed with err:%v\n", err)
-			continue
-		}
-		nodeName := config.Modules.Edged.HostnameOverride
-		if nodeName != pod.Spec.NodeName {
-			fmt.Printf("can't to restart pod: \"%s\" for node: \"%s\"\n", pod.Name, pod.Spec.NodeName)
-			continue
-		}
-		endpoint := config.Modules.Edged.TailoredKubeletConfig.ContainerRuntimeEndpoint
-		remoteRuntimeService, err := remote.NewRemoteRuntimeService(endpoint, time.Second*10, oteltrace.NewNoopTracerProvider())
-
-		var labelSelector = map[string]string{
-			"io.kubernetes.pod.name":      pod.Name,
-			"io.kubernetes.pod.namespace": pod.Namespace,
-		}
-
-		filter := &runtimeapi.ContainerFilter{
-			LabelSelector: labelSelector,
-		}
-		containers, err := remoteRuntimeService.ListContainers(context.TODO(), filter)
-		if err != nil {
-			return err
-		}
-
-		for _, container := range containers {
-			containerID := container.Id
-			err := remoteRuntimeService.StopContainer(context.TODO(), containerID, 3)
-			if err != nil {
-				fmt.Printf("stop containerID:%s with err:%v\n", containerID, err)
-			} else {
-				fmt.Println(containerID)
-			}
-		}
+	kubeClient, err := client.KubeClient()
+	if err != nil {
+		return err
+	}
+	restartResponse, err := podRestart(kubeClient, o.Namespace, podNames)
+	if err != nil {
+		return err
+	}
+
+	for _, errMsg := range restartResponse.ErrMessages {
+		fmt.Println(errMsg)
+	}
+
+	for _, containerID := range restartResponse.ContainerIds {
+		fmt.Println(containerID)
 	}
 	return nil
 }
+
+func podRestart(clientSet *kubernetes.Clientset, namespace string, podNames []string) (*storage.RestartResponse, error) {
+	bodyBytes, err := json.Marshal(podNames)
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.TODO()
```

Comment:
Is it better if the variable context is inputed by the caller?

Reply:
ok